operators/pkg/apis/elasticsearch/v1alpha1/elasticsearch_types.go

```diff
@@ -158,6 +158,13 @@ type ChangeBudget struct {
 	MaxSurge int `json:"maxSurge"`
 }
 
+// AllowsPVCReuse returns true if the current budget allows pods to be deleted
+// in order to reuse their PVC for pods to be created.
+func (c ChangeBudget) AllowsPVCReuse() bool {
+	// only allow pod replacement (deletion first, then creation) if we can go below the target number of pods
+	return c.MaxUnavailable > 0
```

Comment:
I think we eventually need a better way to express that. See https://github.com/elastic/cloud-on-k8s/issues/1153